### PR TITLE
Document netlink extended permissions

### PR DESCRIPTION
**Pull request by tweksteen**

### CODE REVIEW

File: src/xperm_rules.md [hunk 1]
```diff
@@ -135,6 +138,31 @@ Notes:
...
+If the basic permission is granted and no extended permissions are defined for
+the tuple (`src_t`, `tgt_t`, `tclass`), then the access is granted:
+```
+allow src_t tgt_t : netlink_route_socket nlmsg;
+```
+
+Otherwise, it is possible to limit which `nlmsg_type` is accepted for each
+netlink socket class. For example to allow only `RTM_GETROUTE`:
+```
+define(`RTM_GETROUTE', `0x12')
```
Comment (dburgener) [inline src/xperm_rules.md:162]: Is there documentation or kernel headers that we can link to that enumerate all of the netlink message type numbers?

Comment (tweksteen) [reply src/xperm_rules.md:162]: Thanks for the review. I just added the links to the UAPI headers.

**Review by dburgener — APPROVED** [review]

**Comment by dburgener** [comment]
Acked-by: Daniel Burgener <dburgener@linux.microsoft.com>

This looks good to me.  I'd like to wait to merge it until the relevant code changes make it into Linux next and bake there for a few days.  In my opinion, we don't need to wait all the way until they go up to Linus, since this PR clearly references the future version but at least a little linux-next on the off chance that something breaks and they need to be reverted and targeted for a later kernel would be nice.

**Comment by tweksteen** [comment]
Great, that sounds good. Thanks Daniel.

**Comment by tweksteen** [comment]
@dburgener: Linus pulled the change into master last week: https://patchwork.kernel.org/project/linux-security-module/patch/accfdfaee6d92e9049ecc0f8b7e16889@paul-moore.com/

Would it be possible to merge this PR? Thanks

**Comment by dburgener** [comment]
Merged via direct push.  Thanks!

**Pull request closed without merging**